olive/engine/footprint.py

```diff
@@ -68,27 +70,25 @@ def record_objective_dict(self, objective_dict):
         self.objective_dict = objective_dict
 
     def _is_empty_metric(self, metric: FootprintNodeMetric):
-        return not metric or not metric.value
+        return not metric or not metric.value or not metric.value.signal
```

Comment:
Quick question: in which case we could get each `not metric or not metric.value or not metric.value.signal`?

Reply:
During Olive runtime, only check `not metric` is enough since Olive only record metric if the results of evaluation is not None.

But for some case, when we tried to create instance of FootprintNodeMetric manually, it will failed when:
1. `not metric.value`: metric = FootprintNodeMetric()
2. `not metric.value.signal`: metric = FootprintNodeMetric(signal=SignalResult())

Actually, this line is completed by copilot. 🤣  Do you need me to simplify to `not metric`?

Reply:
Yeah it would be better to simplify this if there is no use case of other conditions.